doc/quoting-issues-with-powershell.md

```diff
@@ -0,0 +1,165 @@
+As `az` is a Command Prompt script (at `C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd`), it will have issues when invoked from PowerShell. These issues don't happen when invoking a PowerShell cmdlet:
+
+```powershell
+# Double quotes are preserved
+> Write-Host '"some quoted text"'
+"some quoted text"
+
+# Double quotes are lost
+> python.exe -c "import sys; print(sys.argv[1])" '"some quoted text"'
+some quoted text
+```
+
+## Workaround: the stop-parsing symbol
+To prevent this, you may use [stop-parsing symbol](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_parsing) `--%` between `az` and arguments.
+
+> The stop-parsing symbol (--%), introduced in PowerShell 3.0, directs PowerShell to refrain from interpreting input as PowerShell commands or expressions.
+> When it encounters a stop-parsing symbol, PowerShell treats the remaining characters in the line as a literal.
+
+For instance,
+
+```powershell
+az --% vm create --name xxx
+```
+
+But keep in mind that **the command still needs to be escaped following the Command Prompt syntax**.
+
+```powershell
+# Use --% to stop PowerShell from parsing the argument and escape double quotes
+# following the Command Prompt syntax
+> python.exe --% -c "import sys; print(sys.argv[1])" "\"some quoted text\""
+"some quoted text"
+```
+
+## Typical issues
+
+### Ampersand `&` is interpreted by Command Prompt
+
+ This causes the argument to be parsed again by Command Prompt and breaks the argument. This typically happens when passing a URL with query string to `az`:
+
+```powershell
+> az 'https://graph.microsoft.com/v1.0/me/events?$orderby=createdDateTime&$skip=20' --debug
+az: 'https://graph.microsoft.com/v1.0/me/events?$orderby=createdDateTime' is not in the 'az' command group.
+'$skip' is not recognized as an internal or external command,
+operable program or batch file.
+```
+
+In general, when running `az "a&b"` in PowerShell,
+
+1. Since there is no whitespace in the argument, PowerShell will strip the quotes and pass the argument to Command Prompt
+2. The ampersand `&` is parsed again by Command Prompt as a [command separator](https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-xp/bb490954(v=technet.10)#using-multiple-commands-and-conditional-processing-symbols)
+3. `b` is treated as a separate command by Command Prompt, instead of part of the argument
+
+```powershell
+> az "a&b" --debug
+az: 'a' is not in the 'az' command group.
+'b' is not recognized as an internal or external command,
+operable program or batch file.
+```
+
+This is what `cmd.exe` or Windows system sees:
+
+```cmd
+>az a&b --debug
+az: 'a' is not in the 'az' command group. 
+'b' is not recognized as an internal or external command,
+operable program or batch file.
+```
+
+To solve it:
+
+```powershell
+# Double quotes are parsed by Command Prompt to treat ampersand (&) as a literal character
+> az '"a&b"' --debug
+Command arguments: ['a&b', '--debug']
+
```

Comment:
Based on what I learned from the following section, these also works:
>az "\`"a&b\`"" --debug
>az """a&b""" --debug

Basically, the quotes will first be parsed by Powershell, then be parsed by Command Prompt. If after 2 parses, the double quotes still exist, then it works. This is an alternative to the the `stop-parsing symbol` way. If you could summarize it and put under the Workaround section below `stop-parsing symbol`, it would be very clear to the readers.

Reply:
Thanks for the great suggestion. Add to [here](https://github.com/Azure/azure-cli/pull/13419/files#diff-9d47dca9ba5356e75bb84c36621bd267R21) and [here](https://github.com/Azure/azure-cli/pull/13419/files#diff-9d47dca9ba5356e75bb84c36621bd267R87-R93).